block-layer/codecs/dag-pb.md

```diff
@@ -19,6 +19,7 @@ message PBLink {
   optional string Name = 2;
 
   // cumulative size of target object
+  // deprecated, do not use
```

Comment:
```suggestion
  // Display purposes only ( web gateway directory listing, etc )
  // Historically this field used to hold the on-wire size of the Dag branch
  // It is recommended to either omit this entirely or use the payload-size of the underlying data
```

I agree with @stebalien that not populating this will be too drastic, but for UX reasons. The original value was never useful, and was at best misleading. In my experiments making this equal to the sum of underlying byte payload ( as in what comes back from `cat` ) seemed to work flawlessly and was much easier to reason about: a deep empty directory tree would render as `0`

Reply:
From an implementation point of view populating it but ignoring it is the same as not deprecating it so I'd rather not do that.  OTOH if applications using the implementation choose to ignore it (and they should), that's up to them.

I'd also not like to give implementations a choice of what to use it for because you'll get half doing one thing and half doing the other which helps no-one.

Also, setting it to a different value only works flawlessly because it's not verified.  If it ever is, that'll stop working..

Reply:
I should have been clearer: I agree with the deprecation ( I did so myself - no dag I ever constructed with 3rd party tools had the wire-size listed ).  I disagree with not populating it at all, due to how it is used in the HTTP gateway. We either need to fix the gateway to display the inner node size by interrogating it ( something we will need to do for UFS1.5 anyway ), or we need to clarify that this is a non-validated display-only value which will remain for the lifetime of Unixfs 1.*

Reply:
TLDR: Displaying a long list of 0's on the right side [in contexts like this](https://ipfs.io/ipfs/bafybeif7ceybwlonoct3bnjtnnfookm2iezslo2ypwrsae6ilkc2ucvsfm/linux-5.3.7/) will be a severe UX fail, imho.

Reply:
> We either need to fix the gateway

Thankfully this is something wholly within our power 😄 

> Displaying a long list of 0's on the right side in contexts like this will be a severe UX fail

It would be, but as you quite rightly point out, we're going to have to update the gateway code to pull the root node of directory contents to display metadata anyway, so at that point the `Tsize` property of the DAGLink is redundant so this is not a concern.

We'll also be able to display which links are directories and use any other available information make the UI more informative.